integration-tests/src/test/java/oracle/kubernetes/operator/MiiBaseTest.java

```diff
@@ -76,15 +77,109 @@ public Domain createMiiDomainWithConfigMap(String domainUidPrefix,
+    LoggerHelper.getLocal().log(Level.INFO, "domainName: " + domainName);
+
+    // Get the map of any additional environment vars, or null
+    Map<String, String> additionalEnvMap = (Map<String, String>) domainMap.get("additionalEnvMap");
+    String resultsDir = (String) domainMap.get("resultDir");
+    StringBuffer createDomainImageScriptCmd = new StringBuffer("export WDT_VERSION=");
+
+    createDomainImageScriptCmd.append(BaseTest.WDT_VERSION).append(" && ")
+      .append(getUserProjectsDir())
+      .append("/weblogic-domains/")
+      .append(domainName)
+      .append("/miiWorkDir/")
+      .append("imagetool/bin/imagetool.sh update")
+      .append(" --tag ")
+      .append(imageName)
+      .append(" --fromImage ")
+      .append(domainBaseImageName)
+      .append(" --wdtModel ")
+      .append(resultsDir)
+      .append("/samples/model-in-image/")
+      .append(modelFile)
+      .append(" --wdtVariables ")
+      .append(resultsDir)
+      .append("/samples/model-in-image/")
+      .append(modelPropFile)
+      .append(" --wdtArchive ")
+      .append(getUserProjectsDir())
+      .append("/weblogic-domains/")
+      .append(domainName)
+      .append("/miiWorkDir/models/archive.zip")
+      .append(" --wdtModelOnly ")
+      .append(" --wdtDomainType ")
+      .append(WdtDomainType.WLS.geWdtDomainType());
+
+    // patching the domain
+    LoggerHelper.getLocal().log(Level.INFO, "Command to create domain image: " + createDomainImageScriptCmd);
+    ExecResult result = ExecCommand.exec(createDomainImageScriptCmd.toString(), true, additionalEnvMap);
+    if (result.exitValue() != 0) {
+      throw new RuntimeException(
+        "FAILURE: command "
+          + createDomainImageScriptCmd
+          + " failed, returned "
+          + result.stdout()
+          + "\n"
+          + result.stderr());
+    }
+  }
+
+  /**
+   * Modify the domain yaml to change image name.
+   * @param domainNS the domain namespace
+   * @param domainUid the domain UID
+   * @param imageName image name
+   *
+   * @throws Exception if patching domain fails
+   */
+  protected void modifyDomainYamlWithImageName(
+      Domain domain, String domainNS, String imageName) throws Exception {
+    String versionNo = getRestartVersion(domainNS, domain.getDomainUid());
+    StringBuffer patchDomainCmd = new StringBuffer("kubectl -n ");
+    patchDomainCmd
+        .append(domainNS)
+        .append(" patch domain ")
+        .append(domain.getDomainUid())
+        .append(" --type='json' ")
+        .append(" -p='[{\"op\": \"replace\", \"path\": \"/spec/image\", \"value\": \"'")
+        .append(imageName)
+        .append("'\" }]'");
+
+    // patching the domain
+    LoggerHelper.getLocal().log(Level.INFO, "Command to patch domain: " + patchDomainCmd);
+    ExecResult result = TestUtils.exec(patchDomainCmd.toString());
+    LoggerHelper.getLocal().log(Level.INFO, "Domain patch result: " + result.stdout());
+
+    // verify the domain restarted
```

Comment:
logging is great, thanks!
i also would love to see an assertion in here, and a descriptive error please

Reply:
yea. I have changed it to:

      StringBuffer errorMsg = new StringBuffer("FAILURE: command: ");
      errorMsg
          .append(patchDomainCmd)
          .append(" failed, returned ")
          .append(result.stdout())
          .append("\n")
          .append(result.stderr());

      Assertions.fail(errorMsg + "\n " + ex.getMessage());